packages/desktop-client/src/components/reports/reports/SpendingCard.tsx

```diff
@@ -46,8 +54,36 @@ export function SpendingCard() {
       data.intervalData[todayDay][spendingReportCompare];
   const showLastMonth = data && Math.abs(data.intervalData[27].lastMonth) > 0;
 
+  const spendingReportFeatureFlag = useFeatureFlag('spendingReport');
+
+  if (!spendingReportFeatureFlag) {
+    return (
+      <MissingReportCard isEditing={isEditing} onRemove={onRemove}>
+        The experimental spending report feature has not been enabled.
+      </MissingReportCard>
+    );
+  }
+
   return (
-    <ReportCard to="/reports/spending">
+    <ReportCard
+      isEditing={isEditing}
+      to="/reports/spending"
+      menuItems={[
+        {
+          name: 'remove',
+          text: 'Remove',
+        },
+      ]}
```

Comment:
This seems to get repeated all over, is there a way to avoid that duplication?  Don't they all have a remove, maybe the `ReportCard` can just add a remove to every `menuItems` it gets?

Rename is a bit trickier because at least the way it is now, I don't think you are allowing that for all of them.

Reply:
Similarly maybe the `onMenuSelect` has default behavior to call `onRemove` (which would need passed down) and then fall back to a custom one passed in?

Reply:
The question really is about the prop interface:

```
menuItems: MenuItem[];
onMenuSelect: (item: MenuItem) => void;
```

vs

```
onRemove: () => void;
onRename: () => void;
// .. and eventually could be more operations such as ..
onResize: () => void;
onReset: () => void;
```

--

Personally I'm more inclined towards (1) even though it does have a slight bit of duplication. However, it allows for greater flexibility (i.e. what if we want the ordering of the items to differ for a widget? what if we want different operations? etc). LMK if you disagree. I'm not too fussed on this.

Reply:
I think what I was proposing was the middle ground: you still have menuItems and onMenuSelect, but *also* have onRemove because I think remove is always present.  And possibly onRename (though that isn't always present for some reason).

Reply:
What's the purpose of `menuItems` if we're adding `onRemove` and `onRename`? Just an empty array?

Personally I don't think a mixture of both approaches is the way to go here. We have to choose one and stick with that.

Reply:
I merged the PR to get the ball rolling on other changes to come, but we can continue discussing this here. And if we decide with a different solution - I can do that as a follow-up :)